lib/pengine/pe_output.c

```diff
@@ -3135,54 +3135,103 @@ resource_util_xml(pcmk__output_t *out, va_list args)
     return pcmk_rc_ok;
 }
 
-PCMK__OUTPUT_ARGS("ticket", "pcmk_ticket_t *")
+PCMK__OUTPUT_ARGS("ticket", "pcmk_ticket_t *", "bool", "bool")
 static int
-ticket_html(pcmk__output_t *out, va_list args) {
+ticket_default(pcmk__output_t *out, va_list args) {
     pcmk_ticket_t *ticket = va_arg(args, pcmk_ticket_t *);
+    bool raw = va_arg(args, int);
+    bool details = va_arg(args, int);
 
-    if (ticket->last_granted > -1) {
-        char *epoch_str = pcmk__epoch2str(&(ticket->last_granted), 0);
+    GString *detail_str = NULL;
 
-        out->list_item(out, NULL, "%s:\t%s%s last-granted=\"%s\"",
-                       ticket->id, (ticket->granted? "granted" : "revoked"),
-                       (ticket->standby? " [standby]" : ""),
-                       pcmk__s(epoch_str, ""));
-        free(epoch_str);
-    } else {
-        out->list_item(out, NULL, "%s:\t%s%s", ticket->id,
-                       ticket->granted ? "granted" : "revoked",
-                       ticket->standby ? " [standby]" : "");
+    if (raw) {
+        out->list_item(out, ticket->id, "%s", ticket->id);
+        return pcmk_rc_ok;
     }
 
-    return pcmk_rc_ok;
-}
+    if (details && g_hash_table_size(ticket->state) > 0) {
+        GHashTableIter iter;
+        const char *name = NULL;
+        const char *value = NULL;
+        bool already_added = false;
 
-PCMK__OUTPUT_ARGS("ticket", "pcmk_ticket_t *")
-static int
-ticket_text(pcmk__output_t *out, va_list args) {
-    pcmk_ticket_t *ticket = va_arg(args, pcmk_ticket_t *);
+        detail_str = g_string_sized_new(100);
+        pcmk__g_strcat(detail_str, "\t(", NULL);
 
-    if (ticket->last_granted > -1) {
-        char *epoch_str = pcmk__epoch2str(&(ticket->last_granted), 0);
+        g_hash_table_iter_init(&iter, ticket->state);
+        while(g_hash_table_iter_next(&iter, (void **) &name, (void **) &value)) {
+            if (already_added) {
+                g_string_append_printf(detail_str, ", %s=", name);
+            } else {
+                g_string_append_printf(detail_str, "%s=", name);
+                already_added = true;
+            }
+
+            if (pcmk__str_any_of(name, PCMK_XA_LAST_GRANTED, "expires", NULL)) {
```

Comment:
Does booth create expires? Pacemaker doesn't seem to use it. Might as well create a constant for it while we're here ...